### PR TITLE
Adding Whitesource Scan to Build Workflow

### PR DESCRIPTION
**Pull request by shooshmand-sol**
### What is the purpose of this change?
Adding Whitesource Scan to Build Workflow

### How was this change implemented?
Whitesource scan step added to the build workflow, which does:
- Download latest Mend/Whitesource Agent
- Copy Maven Dependencies which copies all the dependencies to target/lib folder
- Whitesource Scan to run the `wss-unified-agent.jar` scan against `target/lib` folder
- - Solace plugins have been exculded using `WS_EXCLUDES`
- - API key and Project token has been added as repository secrets and passed as env variables.
- 
### How was this change tested?
- https://github.com/SolaceLabs/event-management-agent/actions/runs/4939677815/jobs/8830635052
- on Whitesource: https://saas.whitesourcesoftware.com/Wss/WSS.html#!project;id=9570292;orgToken=0da69ef9-91b1-40a5-933f-2c4b29014240

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ jobs:
       contents: read
       packages: write
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
```
Comment (shooshmand-sol) [inline .github/workflows/build.yaml:26]: upversion the plugin, to address this warning: https://github.com/actions/checkout/issues/1047

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -37,6 +37,16 @@ jobs:
           export SKIP_FLAGS_NON_UNIT_TESTS="-Dcheckstyle.skip -Dpmd.skip -Dcpd.skip -Dfindbugs.skip -Dspotbugs.skip"
           echo "SKIP_FLAGS_NON_UNIT_TESTS=$SKIP_FLAGS_NON_UNIT_TESTS" >> $GITHUB_ENV
           echo "SKIP_FLAGS_ALL_TESTS=$SKIP_FLAGS_NON_UNIT_TESTS -Dmaven.test.skip=true" >> $GITHUB_ENV
+          if [[ $GITHUB_EVENT_NAME != "pull_request" && $GITHUB_REF_NAME=="main" ]]; then
```
Comment (johnvincentcorpuz) [inline .github/workflows/build.yaml:40]: Trying to remember here the case where `GITHUB_EVENT_NAME==pull_request` and `GITHUB_REF_NAME=="main"`. Why did we have to specify `!= pull_request`?

**Review by johnvincentcorpuz — APPROVED** [review]
Looks good to me.

File: .github/workflows/build.yaml (hunk 2 above)
Comment (shooshmand-sol) [reply .github/workflows/build.yaml:40]: Good point, I also realized that `$GITHUB_EVENT_NAME != "pull_request"` is not necessary, for PRs `GITHUB_REF_NAME` vaule is `pr_number/merge`;
so going to drop that!

**Pull request merged**